File: frontend/src/components/FeesPrediction.js
import React, { useState } from 'react';

const FeesPrediction = () => {
  const [injuryType, setInjuryType] = useState('back');
  const [location, setLocation] = useState('urban');
  const [rating, setRating] = useState(4.5);
  const [experience, setExperience] = useState(5);
  const [predictedFee, setPredictedFee] = useState(null);

  const getPredictedFee = async (e) => {
    e.preventDefault();
    const response = await fetch('http://localhost:5000/predict-fees', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        injury_type: injuryType,
        location,
        rating: parseFloat(rating),
        experience: parseInt(experience),
      }),
    });
    const data = await response.json();
    setPredictedFee(data.predicted_fee);
  };

  return (
    <div>
      <h2>Predict Physiotherapy Fees</h2>
      <form onSubmit={getPredictedFee}>
        <label>
          Injury Type:
          <select value={injuryType} onChange={(e) => setInjuryType(e.target.value)}>
            <option value="back">Back</option>
            <option value="knee">Knee</option>
            <option value="shoulder">Shoulder</option>
          </select>
        </label>
        <label>
          Location:
          <select value={location} onChange={(e) => setLocation(e.target.value)}>
            <option value="urban">Urban</option>
            <option value="rural">Rural</option>
          </select>
        </label>
        <label>
          Practitioner Rating:
          <input type="number" step="0.1" value={rating} onChange={(e) => setRating(e.target.value)} />
        </label>
        <label>
          Practitioner Experience (years):
          <input type="number" value={experience} onChange={(e) => setExperience(e.target.value)} />
        </label>
        <button type="submit">Predict Fees</button>
      </form>
      {predictedFee && (
        <div>
          <h3>Predicted Fee: ${predictedFee.toFixed(2)}</h3>
        </div>
      )}
    </div>
  );
};

export default FeesPrediction;

File: backend/fees_prediction.py
import numpy as np
from sklearn.ensemble import RandomForestRegressor

# Sample data: [injury_type (encoded), location (encoded), rating, experience], fee
X = np.array([
    [0, 0, 4.5, 5, 100],
    [1, 1, 4.0, 2, 80],
    [0, 1, 4.8, 10, 150],
    [2, 0, 3.5, 1, 60],
    [1, 0, 4.2, 4, 90],
])
y = X[:, -1]
X = X[:, :-1]

# Injury type encoding: 0: back, 1: knee, 2: shoulder
# Location encoding: 0: urban, 1: rural

model = RandomForestRegressor(n_estimators=100, random_state=42)
model.fit(X, y)

def predict_fees(injury_type, location, rating, experience):
    # Encode input
    injury_encoded = {'back': 0, 'knee': 1, 'shoulder': 2}.get(injury_type, 0)
    location_encoded = {'urban': 0, 'rural': 1}.get(location, 0)

    features = np.array([[injury_encoded, location_encoded, rating, experience]])
    return model.predict(features)[0]
